### PR TITLE
Update hello-world guide

### PR DESCRIPTION
**Pull request by spen**
Hi all!

I saw a PR ping up from @johnmaeda that made an improvement to this guide and noticed that there were other areas that looked like they needed a little updating.

I wanted to address the pre-ES6 vestiges, React.createClass etc. to bring the guide up to the current guideline standards.

I also felt a few slight improvements could be made to the wording & grammar. For the non-obvious I'll add comments inline.

I followed through the guide with the new wording & new code and everything went hunky dory!
Would love to hear some feedback though - this was basically just a skim and update, there may be more fundamental changes in the project that could also be echoed out here that I'm just not aware of!

### CODE REVIEW

File: docs/guide/hello-world.md [hunk 1]
```diff
@@ -146,49 +147,58 @@ import React from 'react';
...
+/**
+ * External dependencies
+ */
+import React from 'react';
+import ReactDom from 'react-dom';
+
+/**
+ * Internal dependencies
+ */
+import HelloWorld from 'my-sites/hello-world/main';
```
Comment (spen) [inline docs/guide/hello-world.md]: I wasn't sure which would be preferred here, import or `require`ing within function as before.

File: docs/guide/hello-world.md [hunk 2]
```diff
@@ -146,49 +147,58 @@ import React from 'react';
...
+export default class HelloWorld extends React.Component {
+	
+};
 ```
 
-Cool. Let's make the React component render something for us. We'll do that by adding a `render()` method that uses the "Main" component and outputs some markup. Inside the `React.createClass` object, after "displayName: 'HelloWorld',", insert before the closing "} );":
+Cool. Let's make the React component render something for us. We'll do that by adding a `render()` method that uses the "Main" component and outputs some markup. Let's add the `render()` method inside of the `React.Component` extension like so:
 
 ```javascript
+export default class HelloWorld extends React.Component {
```
Comment (spen) [inline docs/guide/hello-world.md]: I thought I'd include the extends Component block wrap here for clarity.

**Comment by obenland** [comment]
What's the reason for the change? Providing a description of what's changed and why is always helpful when reviewing PRs

**Comment by spen** [comment]
My bad - I had meant to get in to more detail beyond the the two comments that I left!
I'll get right to it :)

File: docs/guide/hello-world.md [hunk 3]
```diff
@@ -63,15 +63,16 @@ const Controller = {
...
-Next step is to create the main file for your section, called `index.js` within `hello-world`. Run the following command:
+The next step is to create the main file for your section, called `index.js` within `hello-world`.  
+Run the following command to create the file:
 
 ```
 touch client/my-sites/hello-world/index.js
 ```
 
-Here we'll require the `page` module, the My Sites controller, our own controller, and write our main route handler:
+Here we'll require the `page` module, the My Sites controller and our own controller, and write our main route handler:
```
Comment (spen) [inline docs/guide/hello-world.md]: I may be wrong here but an 'and' felt better to me.
'I see two lists here... He're we'll require _some things_ and write our main route handler'

File: docs/guide/hello-world.md [hunk 4]
```diff
@@ -146,49 +147,58 @@ import React from 'react';
...
 
 ```javascript
 helloWorld() {
-	const Main = require( 'my-sites/hello-world/main' );
-
 	// Render hello world...
 	ReactDom.render(
-		React.createElement( Main ),
-		document.getElementById( 'primary' )
+		<HelloWorld />, // Your component
```
Comment (spen) [inline docs/guide/hello-world.md]: I don't think `ReactDom.render`s call is very clear in general so, given this is a hello-world guide, I wanted to just add a little extra clarity :)

File: docs/guide/hello-world.md [hunk 5]
```diff
@@ -146,49 +147,58 @@ import React from 'react';
...
+		document.getElementById( 'primary' ) // Element to render to
 	);
 }
 ```
 
-In the `Main` constant we are getting our main jsx file for our section. We then instruct React to render `Main` in `#primary`, an element that is already on the DOM.
+In the `Main` constant we are getting our main jsx file for our section. We then instruct ReactDom to render `Main` in `#primary`, an element that is already on the DOM.
```
Comment (spen) [inline docs/guide/hello-world.md]: I also just wanted to reenforce: `ReactDom !== React`

Comment (johnmaeda) [inline docs/guide/hello-world.md]: Thanks for these revisions. I'll learn it the new way now (smile).

Comment (spen) [inline docs/guide/hello-world.md]: @johnmaeda NP! Thanks for bringing this doc up in your PR 🙂

**Comment by spen** [comment]
Thanks @obenland! Happy holidays!

**Pull request merged**